Makefile

```diff
@@ -90,6 +101,28 @@ test-silent:
 benchmark-test:
 	. ./scripts/shared_env && go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
 
+test-artifacts: out
+	go test -race -tags integration -o ./out/test-artifacts/unix-engine-tests -c ./agent/engine
+	go test -race -tags integration -o ./out/test-artifacts/unix-stats-tests -c ./agent/stats
+	go test -race -tags integration -o ./out/test-artifacts/unix-app-tests -c ./agent/app
+	go test -tags functional -o ./out/test-artifacts/unix-simple-tests -c ./agent/functional_tests/tests/generated/simpletests_unix/
+	go test -tags functional -o ./out/test-artifacts/unix-handwritten-tests -c ./agent/functional_tests/tests/
+
+	# Race detector on windows requires CGO
+	GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go test -race -tags integration -o ./out/test-artifacts/windows-engine-tests.exe -c ./agent/engine
```

Comment:
Do we need to install mingw32-gcc to run this?  It'd be good to skip these if it's not available.

Reply:
Yep, its in the build dockerfile